okio/src/jvmMain/kotlin/okio/CipherSource.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:JvmMultifileClass
+@file:JvmName("Okio")
+
+package okio
+
+import java.io.IOException
+import javax.crypto.Cipher
+
+private class CipherSource(
+  private val source: BufferedSource,
+  private val cipher: Cipher
+) : Source {
+  constructor(source: Source, cipher: Cipher) : this(source.buffer(), cipher)
```

Comment:
Same . . . just one constructor? The reason I don’t like the two is that `Source` and `BufferedSource` are subtypes of each other so it’s easy to accidentally call the wrong one